misc/docker/init.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/python
+"""
+Init script for dist_test Docker Image
+"""
+
+import os
+import subprocess
+import sys
+
+def log(s):
+    print 'INIT: %s' % s
+
+def run_daemons(daemons):
+    for daemon in daemons:
+        log('Starting daemon: %s' % daemon)
+        rc = subprocess.call(['service', daemon, 'start'])
+        if rc != 0:
+            log('Exiting with status %d..(%s)' % (rc, daemon))
+            sys.exit(rc)
+
+def run_command_and_exit(command):
+    log('Starting command: %s' % command)
+    rc = subprocess.call(command)
+    log('Exiting with status %d..(%s)' % (rc, command))
+    sys.exit(rc)
+
+def get_remaining_args():
+    return sys.argv[1:]
+
+if __name__ == '__main__':
+    daemons = ('mysql', 'beanstalkd')
+    run_daemons(daemons)
+    log('dist_test is installed on /dist_test. Please refer to /dist_test/README.md')
+    log('PLEASE START isolateserver(localhost:4242) MANUALLY (I am not sure to do so :-(')
```

Comment:
@AkihiroSuda 
Ah, so, I've finally gotten some time with one of the original engineers on our current system (which is very hacked together) so I can maybe explain what is going on.

The system as we use it contains three parts:
1. isolate server (used as basically a content-addressable storage only)
2. dist test (the task manager piece in this repo)
3. a client that can talk to the two (slave.py, also in this repo)

When we submit a job we upload the isolate file and related dependencies to the isolate server, then we create a task in the dist_test server.  slaves poll for these tasks and pull them off the work queue.  The isolate server deletes any content not accessed for over 7 days on a daily cron.

So where did our isolate server come from?  Well, we are using an old and modified fork, unfortunately.  We are *pretty sure* this is the revision we are running today:
https://github.com/toddlipcon/luci-go/commit/6f8431521270754084d85b593604ff46f9ac51b1

It contains some "standalone" changes where were not merged upstream, but which might have been implemented by upstream eventually (not sure).  It looks like luci-go added a lot of features since our fork (access control, etc) that might be valuable, so it might be worth your time to try to figure out how to integrate the current version instead, but it isn't going to be "drop in" probably.

Please let me know how it goes either way, as we are likely to be trying to show our setup some love in the near future as well, and please let me know if that clears things up or if you still have any questions.